eodag/plugins/search/qssearch.py

```diff
@@ -1075,8 +1084,46 @@ def do_search(
             ):
                 del prep.total_items_nb
             if items_per_page is not None and len(results) == items_per_page:
-                return results
-        return results
+
+                raw_search_results = self._build_raw_search_results(
+                    results, resp_as_json, kwargs, items_per_page
+                )
+                return raw_search_results
+
+        raw_search_results = self._build_raw_search_results(
+            results, resp_as_json, kwargs, items_per_page
+        )
+        return raw_search_results
+
+    def _build_raw_search_results(self, results, resp_as_json, kwargs, items_per_page):
+        raw_search_results = RawSearchResult(results)
+        raw_search_results.search_params = kwargs | {"items_per_page": items_per_page}
+        links = resp_as_json.get("links") or resp_as_json.get("properties", {}).get(
```

Comment:
this will work only for STAC providers, use `next_page_query_obj_key_path` instead